pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -38,29 +49,231 @@ type Service struct {
 	store             storage.Store
 }
 
+//state defines a didcomm service state
+type state struct {
+	MsgType string            `json:"@type"`
+	ID      string            `json:"@id"`
+	Thread  *decorator.Thread `json:"~thread"`
+	Current string
+}
+
+//key struct prepares the composite key for state persitance and lookup
+type key struct {
+	//protocol is defined in the msg type url for example "didexchange"
+	protocol string
+	//version is derived from msg type as well
+	version string
+}
+
 // New return didexchange service
 func New(store storage.Store, prov provider) *Service {
 	return &Service{outboundTransport: prov.OutboundTransport(), store: store}
 }
 
 // Handle didexchange msg
 func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
-	// TODO add Handle logic
+	//populate request type, protocol, version required for composite key
+	msgType := strings.Split(msg.Type, "/")
+	pk := &key{
+		protocol: msgType[1],
+		version:  msgType[2],
+	}
+	switch msg.Type {
+	case connectionInvite:
+		invitation := &Invitation{}
+		err := json.Unmarshal(msg.Payload, invitation)
+		if err != nil {
+			return err
+		}
+		state := &state{invitation.Type, invitation.ID, nil, ""}
```

Comment:
The thread ID should be set to the invitation.ID here

Reply:
done